eunkyung/app.js

```diff
@@ -34,25 +34,141 @@ app.get('/ping', (req, res) => {
     await mysqlDataSource.query(
         `INSERT INTO users (
             name, 
             password, 
-            email
+            email,
+            profile_image
             ) 
             VALUES (
+                ?,
                 ?,
                 ?,
                 ?
                 );`,
-        [name, password, email]
+        [name, password, email, profile_image]
     )
     res.status(201).json({ message: 'userCreated' })
 })
 
 
 
+app.post('/posting', async (req, res) => {
+    const { title, content, post_image_url, user_id } = req.body
+    await mysqlDataSource.query(
+        `INSERT INTO posts (
+            title,
+            content,
+            post_image_url,
+            user_id
+        )
+        VALUES (
+            ?,
+            ?,
+            ?,
+            ?
+        );`,
+        [title, content, post_image_url, user_id]
+    )
+    res.status(201).json({ message: 'postCreated' })
+})
+
+app.get('/posts', async (req, res) => {
+    await mysqlDataSource.query(
+        `SELECT
+        users.id as userId,
+        users.profile_image as userprofileImage,
+        posts.user_id as postingId,
+        posts.post_image_url as postingImageUrl,
+        posts.content as postingContent 
+        FROM users INNER JOIN posts ON users.id = posts.user_id`
+        , (err, rows) => {
+            res.status(201).json({ data: rows })
+        }
+    )
+})
+
+
+app.get('/posts/:userId', async (req, res) => {
+    const { userId } = req.params;
+
+    await mysqlDataSource.query(
+        `SELECT
+        users.id as userId,
+        users.profile_image as userprofileImage,
+        pp.postings
+        FROM users
+        LEFT JOIN (
+            SELECT 
+            user_id,
+            JSON_ARRAYAGG(
+                JSON_OBJECT(
+                    'postingId', posts.id,
+                    'postingImageUrl', posts.post_image_url,
+                    'postingContent', posts.content
+                ) 
+            ) 
+        as postings
+        FROM posts 
+        GROUP BY user_id  
+        ) pp ON users.id = pp.user_id
+            WHERE users.id = ${userId};`
+        , (err, rows) => {
+            res.status(201).json({ data: rows })
+        }
+    )
+})
+
+app.patch('/edit/:userId/:postId', async (req, res) => {
+    const { userId, postId } = req.params;
+    const { postingContent } = req.body
+
+    await mysqlDataSource.query(
+        `UPDATE posts 
+        SET 
+        posts.content = ? 
+        WHERE posts.user_id = ${userId} AND posts.id = ${postId}
+        `,
+        [postingContent])
+
+
+    await mysqlDataSource.query(
+        `SELECT users.id as userId, users.name as userName, posts.id as postingId, posts.title as postingTitle, posts.content as postingContent FROM users INNER JOIN posts ON users.id = ${userId} AND posts.id = ${postId} WHERE users.id = posts.user_id`
+        , (err, rows) => {
+            res.status(201).json({ data: rows })
+        }
+    )
```

Comment:
아래 수정 내용을 보고 적용해주세요!
- 포매팅 수정
- 두번째 인자로 콜백함수를 쓰지 않고 `?` 매핑을 통해 데이터를 반환하는 방식 참고해주세요!

```suggestion
    const [result] = await mysqlDataSource.query(
        `SELECT
            users.id as userId,
            users.name as userName,
            posts.id as postingId,
            posts.title as postingTitle,
            posts.content as postingContent
          FROM users
          INNER JOIN posts ON users.id = ? AND posts.id = ?
          WHERE users.id = posts.user_id`
        , [ userId, postId ]);
     res.status(200).json({ data: result });
```